class.jetpack.php

```diff
@@ -6967,4 +6967,23 @@ public static function get_file_url_for_environment( $min_path, $non_min_path )
 
 		return plugins_url( $path, JETPACK__PLUGIN_FILE );
 	}
+
+	/**
+	 * Checks for whether Jetpack Rewind is enabled.
+	 * @return bool|int|mixed
+	 */
+	public static function is_rewind_enabled() {
```

Comment:
this is the topic of the conversation :smile:

I _think_ that you are checking the right values here, but we have overlap in terms between the Rewind system and Jetpack itself, so this is the content of the review

_this code_ is checking if a site has "anything to do with Rewind" including that it _could_ have Rewind activated but it's missing the necessary plan.

this _does not_ verify that Rewind is actively backing up and protecting a site, which is probably okay for the purposes we are wanting - know whether or not to prompt to deactivate VaultPress (because if it's missing the plan then VaultPress is probably also missing the plan and if it should be backing up but isn't it's likely due to some external problem)

Reply:
@dmsnell what do you think about setting a transient/option with the response from the /rewind endpoint that we're using for the UI? 

https://github.com/Automattic/jetpack/blob/master/_inc/lib/class.core-rest-api-endpoints.php#L656-L672

Reply:
We could also have it return a different string 'available' or 'enabled' based on its status and test against those.

I don't care which, but we need to decide the conditions for it appearing.

Reply:
So, in EITHER case we don't want VaultPress active, because they should be using rewind.  So the original code is GTG on that front

Reply:
Yeah, the /rewind endpoint we're getting in that link above includes a response that we're using to determine it.  I think it's the way to go.  

```array(
	'reason' => 'host_not_supported',
	'state' => 'active', // Can also be 'available' or 'unavailable'
	'downloads' => array(),
	'last_updated' => 1513366050
);```

Reply:
> `'state' => 'active', // Can also be 'available' or 'unavailable'`

^^^ we will want to make sure we stick to the _actively confusing_ definitions because our terms overlap with different meanings - p9rlnk-1r-p2

> what do you think about setting a transient/option with the response

it's probably about time we look at latency and performance issues and address it appropriately. I doubt putting it in the API call is better than putting it in `get_status()` and `trusted_get_status()` but I don't know if you are talking about a transient on the Jetpack side